google/cloud/aiplatform/tensorboard/uploader_utils.py

```diff
@@ -0,0 +1,297 @@
+                    time_series = ts
+                    num += 1
+                    break
+                if num != 1:
+                    raise ValueError(
+                        "More than one time series resource found with display_name: {}".format(
+                            tag_name
+                        )
+                    )
+            else:
+                raise
+        return time_series
+
+
+class TimeSeriesResourceManager(object):
+    """Helper class managing Time Series resources."""
+
+    def __init__(self, run_resource_id: str, api: TensorboardServiceClient):
+        """Constructor for TimeSeriesResourceManager.
+
+        Args:
+          run_resource_id: The resource id for the run with the following format
+            projects/{project}/locations/{location}/tensorboards/{tensorboard}/experiments/{experiment}/runs/{run}
+          api: TensorboardServiceStub
+        """
+        self._run_resource_id = run_resource_id
+        self._api = api
+        self._tag_to_time_series_proto: Dict[
+            str, tensorboard_time_series.TensorboardTimeSeries
+        ] = {}
+
+    def get_or_create(
+        self,
+        tag_name: str,
+        time_series_resource_creator: Callable[
+            [], tensorboard_time_series.TensorboardTimeSeries
+        ],
+    ) -> tensorboard_time_series.TensorboardTimeSeries:
+        """get a time series resource with given tag_name, and create a new one on
+
+        OnePlatform if not present.
+
+        Args:
+          tag_name: The tag name of the time series in the Tensorboard log dir.
+          time_series_resource_creator: A callable that produces a TimeSeries for
+            creation.
+        """
+        if tag_name in self._tag_to_time_series_proto:
+            return self._tag_to_time_series_proto[tag_name]
+
+        time_series = time_series_resource_creator()
+        time_series.display_name = tag_name
+        try:
+            time_series = self._api.create_tensorboard_time_series(
+                parent=self._run_resource_id, tensorboard_time_series=time_series
+            )
+        except exceptions.InvalidArgument as e:
+            # If the time series display name already exists then retrieve it
+            if "already exist" in e.message:
+                list_of_time_series = self._api.list_tensorboard_time_series(
+                    request=tensorboard_service.ListTensorboardTimeSeriesRequest(
+                        parent=self._run_resource_id,
+                        filter="display_name = {}".format(json.dumps(str(tag_name))),
+                    )
+                )
+                num = 0
+                for ts in list_of_time_series:
+                    time_series = ts
+                    num += 1
+                    break
+                if num != 1:
+                    raise ValueError(
+                        "More than one time series resource found with display_name: {}".format(
```

Comment:
It looks like this would only raise when no time series were found with that display name as the loop breaks after only one iteration.